src/backend/distributed/sql/citus--11.0-4--11.1-1.sql

```diff
@@ -168,3 +168,20 @@ GRANT SELECT ON pg_catalog.pg_dist_background_task_depend TO PUBLIC;
 
 #include "udfs/citus_job_wait/11.1-1.sql"
 #include "udfs/citus_job_cancel/11.1-1.sql"
+
+CREATE TYPE citus.clusterClock AS (logical bigint, counter int);
```

Comment:
>>would it make sense to treat the cluster_clock as a single 64bit integer, bigint in this case?

That was my original code, after seeing funky values in the clock, realized that bigint in SQL is a signed integer

Reply:
It's not just storing the value but also doing arithmetic on this when passing by UDFs, such as `citus_is_clock_after`() etc, that's why it's better to have a well-defined custom datatype. But I will continue to explore that route after the MVP.

Reply:
I was thinking something like this: https://github.com/citusdata/citus/compare/causal-order-clock...causal-order-clock-native

Instead of storing a bitint we actually create a native type that is backed by 8 bytes (uint64) and have corresponding in/out/recv/send functions. My branch contains a quick prototype of the in/out functions, loosely based on `tidin`/`tidout` from postgres.

It probably also needs some operators for ordering since we need a desc index on the clock table catalog etc. Have you tried this approach, and did you run into the same problems you described above?

Reply:
No, I haven't tried this approach, let me take a look.

Reply:
Added a new native cluster_clock type.